File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowels = ['a', 'e', 'i', 'o', 'u']
        start = 0
        end = len(s) - 1
        str = list(s)
        while start <= end:
            if str[start].lower() in vowels:
                if str[end].lower() in vowels:
                    temp = str[start]
                    str[start] = str[end]
                    str[end] = temp
                    start +=1
                end -=1
            else:
                start +=1
        return ''.join(str)

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(s) {
    let vowels = ['a', 'e', 'i', 'o', 'u'];
    let start = 0;
    let end = s.length - 1
    let str = s.split('');
    while(start <= end) {
        if(vowels.includes(str[start].toLowerCase())) {
            if(vowels.includes(str[end].toLowerCase())) {
                let temp = str[start];
                str[start] = str[end];
                str[end] = temp;
                start++;
            }
            end--;
        } else {
            start++;
        }
    }
    return str.join('');
};